docs - AI updates (#74122)
metabase/metabase: docs/ai/mcp.md

If the site URL doesn't match an address your MCP client can reach, like if you're running Metabase in Docker and the site URL got auto-detected from an internal hostname like `metabase-dev:3000`, the client will register but fail the handshake. Your MCP client will typically report a connection failure rather than prompting you to authenticate (for example, Claude Code shows `✗ Failed to connect` rather than `! Needs authentication`).

> ## MCP server requests are handled by the AI client

Reviewer: "AI client" is weirdly both precise and vague.

```suggestion
## MCP server requests don't affect your Metabase AI usage
```

Author: The section is not _just_ about usage - it's also about the fact that the third party AI client the AI part. The intelligence comes from Claude, not Metabase (for inspiration why this is a good point to highlight see the ticket I sent this morning)

Reviewer: The proposed edit implies a third-party AI.

Can we try another heading? 

Another ambiguity to avoid: the "Metabase server requests are handled by." One could say Metabase handles the MCP server requests; AI client makes the requests.

But the proposed edit is also a bit weird because it assumes the misunderstanding it's trying to correct. 

Maybes something like:

```suggestion
## With the MCP server, your client provides the AI
```

> MCP server requests are handled by the AI client - not Metabase. Metabase just handles the MCP call (like searching for an entity or running the query).

Reviewer: ```suggestion
MCP server requests are handled by whatever AI client you're using (like an desktop AI app or editor plugin). The MCP server just provides tools (like searching for an entity or running the query) for your AI.
```

> This means that if you ask Claude "what's our q3 revenue" and instruct it to use Metabase MCP server, _Claude itself_ will determine which tools it needs to use and how - for example, it can decide that it needs to use the tool **construct_query** and **execute_query**, and what those queries might be - and Metabase will just diligently oblige (construct and execute the queries), but nothing more.

Reviewer: ```suggestion
For example, if you ask your AI client to use your Metabase's MCP server "what's our q3 revenue," your client will interact with the MCP server to figure out which tools it needs to field your request. Your AI can decide that it needs to use the tool **construct_query** and **execute_query**, and what those queries might be. Then your client will call those tools for Metabase to run.
```

"Your Metabase's MCP server" is important (the "your" part), as it implies it's just your Metabase, not Metabase the company running a service/routing queries.

In particular, this means that:

- You don't need to have an [AI provider](settings.md#supported-providers) configured in Metabase (because the AI part is handled by your AI client, not Metabase);
- If you do have an AI provider configured in Metabase (for example, for [Metabot](metabot.md)), that provider will _not_ be used for MCP server requestsl

> - MCP server uses the tokens from the third-party AI client you connected to the MCP server, not from the Metabase's AI provider.

Reviewer: ```suggestion
You don't need to have an [AI provider](settings.md#supported-providers) configured in Metabase to use your Metabase's MCP server. If you _do_ have an AI provider configured in Metabase to power Metabot, that provider will _not_ be used for MCP server requests. MCP calls by your local client have no effect on token usage for your Metabase's AI connection.
```

The phrasing implied that MCP servers use tokens.

Also: I wonder if this should be the first paragraph of this section, with the example following.
